lib/discourse_templates/user_extension.rb

```diff
@@ -16,4 +20,20 @@ def can_use_templates?
       guardian.can_see?(category)
     end
   end
+
+  def can_use_private_templates?
+    return false unless SiteSetting.discourse_templates_enable_private_templates
+    return false unless SiteSetting.tagging_enabled
+    return false if SiteSetting.discourse_templates_private_templates_tags.blank?
+    return true if guardian.is_staff?
+
+    allowed_groups_ids = SiteSetting.discourse_templates_groups_allowed_private_templates&.split("|")&.map(&:to_i)
+
+    allowed_groups_ids.any? do |group_id|
+      return false if group_id == 0
+
+      # the user can use templates if belongs to at least one of the allowed groups
+      GroupUser.where(group_id: group_id, user_id: self.id).exists?
```

Comment:
I think you can just do this:

```suggestion
      GroupUser.exists?(group_id: group_id, user_id: self.id)
```